ctadata/api.py

```diff
@@ -1,24 +1,36 @@
+import httpx
 import logging
 import os
 import requests
 import time
 from .util import urljoin_multipart
-from . import __version__
+from webdav4.client import Client
 
+import importlib.metadata
+__version__ = importlib.metadata.version("ctadata")
 
 logger = logging.getLogger(__name__)
 
 
 class StorageException(Exception):
     pass
 
+
+class CertificateError(Exception):
+    def __init__(self, message="invalid certificate"):
```

Comment:
are you sure this adds anything to parent behavior? It stores mesasage in ".message" instead of '.args' but this is never used like that.

Reply:
So, the init call allows having a proper error message displayed when an exception is triggered + the variable allows the user to retrieve the error message if needed.

Reply:
Is this really different?

```python
>>> class CertificateErrorBare(Exception): pass

>>> raise CertificateErrorBare("proper message")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
__main__.CertificateErrorBare: proper message

>>> raise CertificateError("proper message")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
__main__.CertificateError: proper message
>>> 

>>> CertificateErrorBare("proper message").args
('proper message',)

>> CertificateError("proper message").message
'proper message'
>> CertificateError("proper message").args
('proper message',)
```

It's displayed in the same way, actually `.message` is not displayed since you did not override any repr or so, and what's  displayed comes from `.args` (so in fact the same message is stored in two attributes).

Or is `.message` attribute used here by anything in particular? The only advantage of copying message text into an attribute is that it becomes clear that Exception.args[0] is in fact a message. Which is fine, but clear for whom?..

Reply:
I can simplify this code then 👍

Reply:
Just wanted to know what you meant maybe I missed something! :) 
I know that plenty of Exception children use .message attribute, but in fact Exception itself treats its .args as message, from the way it prints it for example. The [doc](https://docs.python.org/3/library/exceptions.html#base-classes) does say explicitly that it's common to put a message there.